src/Microsoft.Azure.ServiceBus/SessionReceivePump.cs

```diff
@@ -122,7 +122,13 @@ async Task SessionPumpTaskAsync()
                     concurrentSessionSemaphoreAcquired = true;
 
                     await this.maxPendingAcceptSessionsSemaphoreSlim.WaitAsync(this.pumpCancellationToken).ConfigureAwait(false);
-                    var session = await this.client.AcceptMessageSessionAsync().ConfigureAwait(false);
+
+                    IMessageSession session;
+                    if(string.IsNullOrEmpty(sessionHandlerOptions.SessionId))
```

Comment:
There's no need in `if/else`. `null` indicates any session id. Therefore, `sessionHandlerOptions.SessionId` can be passed for both cases.

Reply:
Allright, I’ll Change that tomorrow.

Reply:
Changed it. Check latest commit.